app/routes/applications/upload.tsx

```diff
@@ -5,17 +5,22 @@ import {
   redirect,
 } from "@remix-run/node";
 import { Form, useLoaderData } from "@remix-run/react";
-import { getAccessToken } from "~/cookies.server";
+import { getAccessToken } from "~/token.server";
 
 import { applicationByName, submitJob } from "~/models/applicaton.server";
+import { getLevel } from "~/models/user.server";
 
 export const loader = async ({ request }: LoaderArgs) => {
   const name = "haddock3";
   const app = await applicationByName(name);
-  const access_token = await getAccessToken(request);
-  if (access_token === undefined) {
+  const accessToken = await getAccessToken(request);
+  if (accessToken === undefined) {
     throw new Error("Unauthenticated");
   }
+  const level = await getLevel(accessToken)
+  if (level === '') {
```

Comment:
Wouldn't it be better to check explicitly if level is in allowed roles?

Reply:
check moved to own method isSubmitAllowed